File: since2020/LeetCode/LeetCode 442.cpp
/**
 *  @judge LeetCode
 *  @id 442
 *  @name Find All Duplicates in an Array
 * 
 *  @tag Array
 */
class Solution {
public:
	vector<int> findDuplicates(vector<int>& nums) {
		vector<int> res;
		for (int i = 0; i < nums.size(); i++) {
			int j = abs(nums[i]) - 1;
			if (nums[j] < 0)
				res.push_back(j + 1);
			else
				nums[j] *= -1;
		}
		return res;
	}
};

File: since2020/LeetCode/LeetCode 442.py
'''
	@judge LeetCode
	@id 442
	@name Find All Duplicates in an Array

	@tag Array
'''
class Solution:
	def findDuplicates(self, nums: List[int]) -> List[int]:
		res = []
		for idx in range(len(nums)):
			jdx = abs(nums[idx]) - 1
			if nums[jdx] < 0:
				res.append(jdx + 1)
			else:
				nums[jdx] *= -1
		return res